docs/source/notes/autograd.rst

```diff
@@ -210,3 +210,79 @@ No thread safety on C++ hooks
 Autograd relies on the user to write thread safe C++ hooks. If you want the hook
 to be correctly applied in multithreading environment, you will need to write
 proper thread locking code to ensure the hooks are thread safe.
+
+Autograd for Complex Numbers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+What notion of complex derivative does PyTorch use?
+***************************************************
+
+   PyTorch follows `_JAX's <https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Complex-numbers-and-differentiation>`
+   convention for autograd for Complex Numbers.
+
+    For a function :math:`F: C → C`
+
+    .. math::
+        def F(z):
+            x, y = real(z), imag(z)
+            return u(x, y) + v(x, y) * 1j
+
+    The JVP and VJP for function :math:`F` are defined as:
+
+    .. math::
+
+        J = \begin{bmatrix}
+            \partial_0u(x, y) & \partial_1u(x, y)\\
+            \partial_0v(x, y) & \partial_1v(x, y) \end{bmatrix}
+
+        def JVP(tangent):
+            c, d = real(tangent), imag(tangent)
+            return \begin{bmatrix} 1 & i \end{bmatrix} * J * \begin{bmatrix} c \\ d \end{bmatrix}
+
+        def VJP(cotangent):
+            c, d = real(cotangent), imag(cotangent)
+            return \begin{bmatrix} c & -d \end{bmatrix} * J * \begin{bmatrix} 1 \\ -i \end{bmatrix}
+
+What happens if I call backward() on a complex scalar?
+******************************************************
+
+    1. For holomorphic functions, you get the same result as expected from using Cauchy-Riemann equations.
+    2. For non-holomorphic functions, the partial derivatives of :math:`v(x, y)` are discarded.
```

Comment:
of the imaginary part of the function (`v(x, y)` above) are discarded (e.g., this is equivalent to dropping the imaginary part of the loss before performing a backwards). To get the gradient with respect to the imaginary components of the function, you must explicitly specify `gradient=torch.tensor(1j)`)

Reply:
yeah that's easier to read. updated!
not sure what you mean by:
```
To get the gradient with respect to the imaginary components of the function, you must explicitly specify gradient=torch.tensor(1j))
```

do you mean to say for any other desired behavior, specify the `grad_out` accordingly?